### PR TITLE
Ecm/revision

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description
Create config to enable testing of newer API revisions more easily

# Check List

- [x] Are you changing anything with the public API?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you tested this change on real device?
- [x] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all Android versions the SDK currently supports?

## Changelog / Code Overview

Made the API revision configurable

## Test Plan

Update internal test app to allow configuring the revision, much like we have base url configurable for local/dev environments

## Related Issues/Tickets

#189 
CHNL-12477

### CODE REVIEW

File: sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/BaseApiRequestTest.kt [hunk 1]
```diff
@@ -16,7 +16,7 @@ internal abstract class BaseApiRequestTest<T> : BaseTest() where T : KlaviyoApiR
     open val expectedHeaders = mapOf(
         "Content-Type" to "application/json",
         "Accept" to "application/json",
-        "Revision" to "2023-07-15",
+        "Revision" to "1234-56-78",
```
Comment (dan-peluso) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/BaseApiRequestTest.kt:19]: bring me back to the 1200s

**Review by dan-peluso — APPROVED** [review]
LGTM 🚢

File: sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/BaseApiRequestTest.kt (hunk 1 above)
Comment (evan-masseau) [reply sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/BaseApiRequestTest.kt:19]: Don't worry, our [API revision policies](https://developers.klaviyo.com/en/docs/api_versioning_and_deprecation_policy#fall-forward-behavior) will fall forward 790 years

**Review by njparadis — APPROVED** [review]
LGTM godspeed Kotlin kings 👑

**Pull request merged**